metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/DataSourceManager.java

```diff
@@ -50,13 +50,13 @@ public static DataSourceManager get() {
     /**
      * Initialize a data source and store it.
      *
-     * @param catalogName catalog name
+     * @param name catalog name
      * @param properties  properties
      * @return DataSourceManager
      */
-    public DataSourceManager load(final String catalogName, final Map<String, String> properties) {
-        if (dataSources.get(catalogName) == null) {
-            createDataSource(catalogName, properties);
+    public DataSourceManager load(final String name, final Map<String, String> properties) {
+        if (dataSources.get(name) == null) {
```

Comment:
Should this be synchronized instead of synchronizing the method below? that way you avoid having two threads waiting on that method just to do the check again?

Reply:
No, you always want to synchronize at a level closest to the update of the critical data. Putting synchronized in this method also would make the threads wait to get the lock.